### PR TITLE
CronSchedule - add timezone support. Refactor using LocalDateTime

### PR DESCRIPTION
**Pull request by jlhughes**

### CODE REVIEW

File: src/foam/nanos/cron/test/CronScheduleTest.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
```
Comment (kgrgreer) [inline src/foam/nanos/cron/test/CronScheduleTest.js]: 2023

File: src/foam/nanos/cron/CronSchedule.js [hunk 2]
```diff
@@ -54,14 +59,19 @@ foam.CLASS({
       name: 'dayOfWeek',
       value: -1,
       documentation: `Day of week to execute script.
-          Ranges from 0 - 6, where 0 is Sunday. -1 for wildcard`
+          Ranges from 1 - 7, where 1 is Monday (matches java time dayofweek). -1 for wildcard`
```
Comment (kgrgreer) [inline src/foam/nanos/cron/CronSchedule.js:62]: Do we need to migrate any old data or adapt so that 0 is converted to 7?

Comment (jlhughes) [reply src/foam/nanos/cron/CronSchedule.js:62]: "Admin Access Configuration" running only on ablii.

File: src/foam/nanos/cron/CronSchedule.js [hunk 3]
```diff
@@ -79,53 +89,69 @@ foam.CLASS({
...
+  last = LocalDate.now(zone).atStartOfDay();
+} else {
+  last = LocalDateTime.ofInstant(from.toInstant(), zone);
+}
+var time = last;
+
+// test if component needs to be bumped to the next hour, minute,...
+if ( getHour() > -1 ) {
+  if ( time.getHour() >= getHour() &&
+       ! time.isAfter(last) ) {
```
Comment (adamfox-nanopay) [inline src/foam/nanos/cron/CronSchedule.js:109]: time = last so not sure you need this check

Comment (jlhughes) [reply src/foam/nanos/cron/CronSchedule.js:109]: Yes, this particular test is not needed. 
Presently all the test conditions are the same and I feel I'd want to comment why the first one is different. 
I would like to leave it, but you make the call.

**Review by adamfox-nanopay — APPROVED** [review]

**Pull request merged**